### PR TITLE
Improved cheeck-= and check-within error messages

### PR DESCRIPTION
**Pull request by matteo-daddio**
Improved documentation and error messages of `check-=` and `check-within` as discussed in issue #165.

### CODE REVIEW

**Comment by matteo-daddio** [comment]
Improved documentation and error messages of `check-= `and `check-within` as discussed in issue https://github.com/racket/rackunit/issues/165.

File: rackunit-doc/rackunit/scribblings/check.scrbl [hunk 1]
```diff
@@ -74,40 +74,46 @@ The following check fails:
...
-  (check-within (hash 'C 20 'F 68) (hash 'C 25 'F 77) 10)
+  (check-within (list 6.02 9.99) (list 6 10) 0.05)
+  (check-within (flvector 3.01 4.01 5.014) (flvector 3.0 4.0 5.0) 0.02)
+  (check-within (hash 'C 25 'F 77) (hash 'C 20 'F 68) 10)
 ]
+
 And the following checks fail:
 @interaction[#:eval rackunit-eval
-  (check-within (list 6.0e23 10.0) (list 6.02e23 9.8) 0.05)
+  (check-within (list 6.02e23 9.8) (list 6.0e23 10.0) 0.05)
```
Comment (sorawee) [inline rackunit-doc/rackunit/scribblings/check.scrbl:116]: Could you revert these lines? I think the intention is to test against the Avogadro constant and the gravity constant of Earth, which are 6.02e23 and 9.8 respectively, so these should be the expected values.

Comment (matteo-daddio) [reply rackunit-doc/rackunit/scribblings/check.scrbl]: Did it. What do you think?

File: rackunit-doc/rackunit/scribblings/check.scrbl [hunk 2]
```diff
@@ -74,40 +74,51 @@ The following check fails:
...
 ]
+
 And the following checks fail:
 @interaction[#:eval rackunit-eval
-  (check-within (list 6.0e23 10.0) (list 6.02e23 9.8) 0.05)
+  (define (avogadro-constant) 6.0e23) ;computes the Avogadro constant
+  (define (gravity-constant)  10.0)   ;computes the Gravity constant
+  
+  (check-within (list (avogadro-constant) (gravity-constant))
+                (list 6,02214076e23 9.81) 0.05)
```
Comment (sorawee) [inline rackunit-doc/rackunit/scribblings/check.scrbl:120]: Should be `.`, not `,`.

File: rackunit-doc/rackunit/scribblings/check.scrbl [hunk 3]
```diff
@@ -74,40 +74,51 @@ The following check fails:
...
+This check is similar to @racket[check-=] except it works on data structures
+like lists, flvectors and hash tables.
 
 For example, the following checks pass:
 
 @interaction[#:eval rackunit-eval
-  (check-within (list 6 10) (list 6.02 9.99) 0.05)
-  (check-within (flvector 3.0 4.0 5.0) (flvector 3.01 4.01 5.014) 0.02)
-  (check-within (hash 'C 20 'F 68) (hash 'C 25 'F 77) 10)
+  (check-within (list 6.02 9.99) (list 6 10) 0.05)
```
Comment (sorawee) [inline rackunit-doc/rackunit/scribblings/check.scrbl:109]: Here, too.

Comment (matteo-daddio) [reply rackunit-doc/rackunit/scribblings/check.scrbl]: Corrected.

Comment (matteo-daddio) [reply rackunit-doc/rackunit/scribblings/check.scrbl]: I think that the tests are ok.

**Comment by sorawee** [comment]
There are more changes I'd like to make, but they are just nits, so I'll go ahead and merge this, and make the change later. In particular, I'd like to keep similarity between "For example, the following checks pass:" and "And the following checks fail:" as much as possible, so that readers can contrast them easily. That means:
- `check-within`for `hash` should have `(hash 'C 25 'F 77)` as the expected value. 
- `check-within` for `list` should have the same expected value.

**Comment by matteo-daddio** [comment]
Thank you very much sorawee!

Matteo d’Addio
________________________________
Da: sorawee ***@***.***>
Inviato: Saturday, November 4, 2023 3:23:04 PM
A: racket/rackunit ***@***.***>
Cc: matteo-daddio ***@***.***>; Author ***@***.***>
Oggetto: Re: [racket/rackunit] Improved cheeck-= and check-within error messages (PR #166)


Merged #166<https://github.com/racket/rackunit/pull/166> into master.

**Pull request merged**